src/figuresClasses.ts

```diff
@@ -1,19 +1,77 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Figure['shape'] = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides = [this.a, this.b, this.c];
+    const longestSide = Math.max(this.a, this.b, this.c);
+
+    const twoOther = sides.filter((side) => side !== longestSide)
+      .reduce((side, sum) => side + sum);
+
+    if (twoOther <= longestSide || sides.some((num) => num <= 0)) {
+      throw new Error('Not valid data');
+    }
```

Comment:
In your code, 2 different errors display the same message. These are different errors.
Use more descriptive messages for errors
```suggestion
    const sumOfTwoLowerSides = sides
      .filter((side) => side !== longestSide)
      .reduce((side, sum) => side + sum);

    if (sides.some((num) => num <= 0)) {
      throw new Error('All sides have to be greater than zero');
    }
    if (sumOfTwoLowerSides <= longestSide) {
      throw new Error('The longest side of a triangle is >= than a sum of two others',);
    }
```